src/zero/module/aou.clj

```diff
@@ -1,23 +1,385 @@
+                                        :reported_gender "Female",
+                                        :cluster_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_CEPH_A.egt",
+                                        :sample_lsid "broadinstitute.org:bsp.dev.sample:NOTREAL.NA12878",
+                                        :sample_alias "NA12878",
+                                        :green_idat_cloud_path "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/idats/7991775143_R01C01/7991775143_R01C01_Grn.idat",
+                                        :params_file "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/inputs/7991775143_R01C01/params.txt",
+                                        :gender_cluster_file "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_gender.egt"}))
+
+#_(defn update-workload!
+    "Use transaction TX to update WORKLOAD statuses."
+    [tx workload])
 
 (defn add-workload!
-  "Add the workload described by BODY to the database DB."
+  "Use transaction TX to add the workload described by BODY to the database DB.
+   Due to the continuous nature of the AoU dataflow, this function will only
+   create a new workload table if it does not exist otherwise append records
+   to the existing one."
   [tx body]
-  (->> body
-       first
-       (filter second)
-       (into {})))
+  (let [{:keys [creator cromwell pipeline project]} body
+        {:keys [release top]} workflow-wdl
+        {:keys [commit version]} (zero/get-the-version)
+        probe-result (jdbc/query tx ["SELECT * FROM workload WHERE project = ? AND pipeline = ? AND release = ?"
+                                     project top release])]
+    (if (seq probe-result)
+      ;; if a table already exists, we return its uuid and the name
+      (let [[{:keys [id uuid]}] probe-result
+            table (format "%s_%09d" pipeline id)]
+        [uuid table])
+      ;; if the expected table doesn't exist, we create record + table and return uuid and name
+      (let [[{:keys [id uuid]}]
+            (jdbc/insert! tx :workload {:commit   commit
+                                        :creator  creator
+                                        :cromwell cromwell
+                                        :input    "aou-inputs-placeholder"
+                                        :output   "aou-outputs-placeholder"
+                                        :project  project
+                                        :release  release
+                                        :uuid     (UUID/randomUUID)
+                                        :version  version
+                                        :wdl      top})
+            table (format "%s_%09d" pipeline id)
+            table_seq (format "%s_id_seq" table)
+            kind  (format (str/join " " ["UPDATE workload"
```

Comment:
You can use a prepared statement here: https://github.com/broadinstitute/wfl/blob/ec7cc6bf0f68fa4d81f9499796bc3bbece57f42e/src/zero/module/all.clj#L105

Reply:
Thanks! I'm wondering how could we apply prepared statement for something like `(jdbc/db-do-commands tx [kind idx work link-idx-work])` that requires a sequence of sql executions, would we thread them in a seq of `(jdbc/execute! foo)`s?

Reply:
That seems like it would work! There is also `(jdbc/db-do-prepared foo)` which looks like the prepared statement equivalent to `(jdbc/db-do-commands foo)`

Reply:
`jdbc/db-do-prepared` looks cool! Let me take a look 👀 !